internal/dataplane/parser/translate_ingress_test.go

```diff
@@ -737,160 +735,371 @@ func TestFromIngressV1(t *testing.T) {
-			IngressesV1: []*netv1.Ingress{
-				ingressList[9],
-			},
+			parsedInfo := p.ingressRulesFromIngressV1()
+			assert.Equal(t, 1, len(parsedInfo.ServiceNameToServices))
+			assert.Equal(t, "foo-svc.foo-namespace.80.svc", *parsedInfo.ServiceNameToServices["foo-namespace.foo.foo-svc.80"].Host)
+			assert.Equal(t, 80, *parsedInfo.ServiceNameToServices["foo-namespace.foo.foo-svc.80"].Port)
+
+			assert.Equal(t, "/", *parsedInfo.ServiceNameToServices["foo-namespace.foo.foo-svc.80"].Routes[0].Paths[0])
+			assert.Equal(t, "example.com", *parsedInfo.ServiceNameToServices["foo-namespace.foo.foo-svc.80"].Routes[0].Hosts[0])
 		})
-		require.NoError(t, err)
-		p := mustNewParser(t, store)
+		t.Run("ingress rule with default backend", func(t *testing.T) {
+			store, err := store.NewFakeStore(store.FakeObjects{
+				IngressesV1: []*netv1.Ingress{
+					ingressList[0],
+					ingressList[2],
+				},
+			})
+			require.NoError(t, err)
+			p := setupParser(t, store)
 
-		parsedInfo := p.ingressRulesFromIngressV1()
-		_, ok := parsedInfo.ServiceNameToServices["foo-namespace.foo-svc.pnum-80"]
-		assert.True(ok)
-	})
-	t.Run("Ingress rule with regex prefixed path creates route with Kong regex prefix", func(t *testing.T) {
-		store, err := store.NewFakeStore(store.FakeObjects{
-			IngressesV1: []*netv1.Ingress{
-				ingressList[9],
-			},
+			parsedInfo := p.ingressRulesFromIngressV1()
+			assert.Equal(t, 2, len(parsedInfo.ServiceNameToServices))
+			assert.Equal(t, "foo-svc.foo-namespace.80.svc", *parsedInfo.ServiceNameToServices["foo-namespace.foo.foo-svc.80"].Host)
+			assert.Equal(t, 80, *parsedInfo.ServiceNameToServices["foo-namespace.foo.foo-svc.80"].Port)
+
+			assert.Equal(t, "/", *parsedInfo.ServiceNameToServices["foo-namespace.foo.foo-svc.80"].Routes[0].Paths[0])
+			assert.Equal(t, "example.com", *parsedInfo.ServiceNameToServices["foo-namespace.foo.foo-svc.80"].Routes[0].Hosts[0])
+
+			assert.Equal(t, 1, len(parsedInfo.ServiceNameToServices["bar-namespace.default-svc.80"].Routes))
+			assert.Equal(t, "/", *parsedInfo.ServiceNameToServices["bar-namespace.default-svc.80"].Routes[0].Paths[0])
+			assert.Equal(t, 0, len(parsedInfo.ServiceNameToServices["bar-namespace.default-svc.80"].Routes[0].Hosts))
 		})
-		require.NoError(t, err)
-		p := mustNewParser(t, store)
+		t.Run("ingress rule with TLS", func(t *testing.T) {
+			store, err := store.NewFakeStore(store.FakeObjects{
+				IngressesV1: []*netv1.Ingress{
+					ingressList[1],
+				},
+			})
+			require.NoError(t, err)
+			p := setupParser(t, store)
 
-		parsedInfo := p.ingressRulesFromIngressV1()
-		assert.Equal(translators.KongPathRegexPrefix+"/foo/\\d{3}", *parsedInfo.ServiceNameToServices["foo-namespace.foo-svc.pnum-80"].Routes[0].Paths[0])
+			parsedInfo := p.ingressRulesFromIngressV1()
+			assert.Equal(t, 2, len(parsedInfo.SecretNameToSNIs.Hosts("bar-namespace/sooper-secret")))
+			assert.Equal(t, 2, len(parsedInfo.SecretNameToSNIs.Hosts("bar-namespace/sooper-secret2")))
```

Comment:
Why no `assert.Len()` 🤔 ?

Reply:
A matter of copy-pasta from existing tests - thanks for noticing. I've updated all of them to use `assert.Len` :)